Explorer/Assets/DCL/Multiplayer/Movement/Systems/SDKAvatarShapesMotionSystem.cs

```diff
@@ -0,0 +1,108 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Character.Components;
+using DCL.CharacterMotion.Animation;
+using DCL.CharacterMotion.Components;
+using DCL.Diagnostics;
+using DCL.Profiles;
+using ECS.Abstract;
+using UnityEngine;
+
+namespace DCL.Multiplayer.Movement.Systems
+{
+    [UpdateInGroup(typeof(PresentationSystemGroup))]
+    [LogCategory(ReportCategory.AVATAR)]
+    public partial class SDKAvatarShapesMotionSystem : BaseUnityLoopSystem
+    {
+        private const float ROTATION_SPEED = 5f;
+        private const float RUN_SPEED_THRESHOLD = 9.5f;
+        private const float JOG_SPEED_THRESHOLD = 4f;
+
+        private SDKAvatarShapesMotionSystem(World world) : base(world) { }
+
+        protected override void Update(float t)
+        {
+            UpdateMotionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(Profile))]
+        private void UpdateMotion(
+            in IAvatarView view,
+            ref CharacterTransform characterTransformComponent,
+            ref CharacterLastPosition characterLastPositionComponent,
+            ref CharacterAnimationComponent animationComponent)
+        {
+            UpdateRotation(characterTransformComponent, ref characterLastPositionComponent);
+            UpdateAnimations(view, characterTransformComponent, ref characterLastPositionComponent, ref animationComponent);
+            UpdateLastPosition(ref characterTransformComponent, ref characterLastPositionComponent);
+        }
+
+        private static void UpdateRotation(
+            CharacterTransform characterTransformComp,
+            ref CharacterLastPosition characterLastPositionComp)
+        {
+            var currentPosition = characterTransformComp.Transform.position;
+            var oldPosition = characterLastPositionComp.LastPosition;
+
+            if (oldPosition == currentPosition)
+                return;
+
+            Vector3 newDirection = currentPosition - oldPosition;
+            newDirection.y = 0;
+
+            if (newDirection != Vector3.zero)
+            {
+                Vector3 currentForward = characterTransformComp.Transform.forward;
+                Vector3 targetForward = newDirection.normalized;
+                characterTransformComp.Transform.forward = Vector3.Lerp(currentForward, targetForward, UnityEngine.Time.deltaTime * ROTATION_SPEED);
+            }
+        }
+
+        private static void UpdateAnimations(
+            IAvatarView view,
+            CharacterTransform characterTransformComp,
+            ref CharacterLastPosition characterLastPositionComp,
+            ref CharacterAnimationComponent animationComponent)
+        {
+            const bool IS_JUMP_TRIGGERED = false;
+            const bool IS_STUNNED = false;
```

Comment:
i think you can remove (inline) these constants and use parameter name in the function:
            ```AnimationStatesLogic.SetAnimatorParameters(view, ref animationComponent.States, animationComponent.States.IsJumping, jumpTriggered: false, isStunned: false);```

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/3297/commits/157b8e8a3ebf9f90d45a78f1c67b20663ecea492